### PR TITLE
Improve impression logging for experimental campaigns

### PR DESCRIPTION
**Pull request by JasonAHeron**

### CODE REVIEW

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java [hunk 1]
```diff
@@ -197,7 +197,14 @@ private boolean shouldIgnoreCache(String event) {
                                       Logging.logi(
                                           String.format(
                                               "Already impressed %s ? : %s",
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java]: Lets do `Already impression campaign / experiment` appropriately... Also, lets move this log line generation to a function.

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/model/ProtoMarshallerClient.java [hunk 2]
```diff
@@ -207,11 +208,14 @@ private static Text decode(MessagesProto.Text in) {
   /** Tranform {@link MessagesProto.Content} proto to an {@link InAppMessage} value object */
   public static InAppMessage decode(
       @Nonnull MessagesProto.Content in,
-      String campaignId,
-      String campaignName,
-      boolean isTestMessage,
+      @NonNull String campaignId,
+      @NonNull String campaignName,
+      @NonNull boolean isTestMessage,
```
Comment (PromanSEW) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/model/ProtoMarshallerClient.java]: `boolean` primitive cannot be null

Comment (JasonAHeron) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/model/ProtoMarshallerClient.java]: Thanks!

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DisplayCallbacksImpl.java [hunk 3]
```diff
@@ -224,9 +224,9 @@ private void logActionNotTaken(String action) {
   }
 
   private Completable logToImpressionStore() {
-    Logging.logd("Attempting to record: message impression in impression store");
-    String campaignId = inAppMessage.getCampaignId();
-
+    String campaignId = inAppMessage.getCampaignMetadata().getCampaignId();
+    Logging.logd(
+        "Attempting to record message impression in impression store for id:" + campaignId);
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/DisplayCallbacksImpl.java]: nit: add a space after ":"

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ImpressionStorageClient.java [hunk 4]
```diff
@@ -84,7 +85,11 @@ private void clearInMemCache() {
   }
 
   /** Returns {@code Single.just(true)} if the campaign has been impressed */
-  public Single<Boolean> isImpressed(String campaignId) {
+  public Single<Boolean> isImpressed(CampaignProto.ThickContent content) {
+    String campaignId =
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ImpressionStorageClient.java]: nit: In experimental payload case, this is technically an `experimentId`, right? Should add that as a comment / clarification and / or rename this variable?

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java [hunk 5]
```diff
@@ -288,6 +282,20 @@ private boolean shouldIgnoreCache(String event) {
...
+  private static void logImpressionStatus(ThickContent content, Boolean isImpressed) {
+    if (content.getPayloadCase().equals(ThickContent.PayloadCase.VANILLA_PAYLOAD)) {
+      Logging.logi(
+          String.format(
+              "Already impressed campaign %s ? : %s",
+              content.getVanillaPayload().getCampaignName(), isImpressed));
+    } else if (content.getPayloadCase().equals(ThickContent.PayloadCase.EXPERIMENTAL_PAYLOAD)) {
+      Logging.logi(
+          String.format(
+              "Already impressed campaign %s ? : %s",
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java]: s/campaign/experiment?

File: firebase-inappmessaging/src/test/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManagerTest.java [hunk 6]
```diff
@@ -560,7 +559,7 @@ public void stream_onCacheWriteFailure_AbsorbsError() {
 
   @Test
   public void stream_whenCampaignImpressed_filtersCampaign() {
-    when(impressionStorageClient.isImpressed(anyString())).thenReturn(Single.just(true));
+    when(impressionStorageClient.isImpressed(any())).thenReturn(Single.just(true));
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/test/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManagerTest.java]: why did we change this? The more specific it is, the better. I'd vote for bringing this back.

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ImpressionStorageClient.java (hunk 4 above)
Comment (JasonAHeron) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ImpressionStorageClient.java]: Yeah, throughout the rest of the code we call this a campaignId and basically what it really is is a unique message identifier. In reality it should probably just be ID. it doesn't even matter that it's a campaignId or an experimentId just that it uniquely identifies this message. so I've just left it was campaignId.

**Pull request merged**